src/NBitcoin/Network.cs

```diff
@@ -10,1429 +12,1079 @@
-		{
-			get
-			{
-				return this.vSeeds;
-			}
-		}
-#endif
-		public byte[] _MagicBytes;
-		public byte[] MagicBytes
-		{
-			get
-			{
-				if(_MagicBytes == null)
-				{
-					var bytes = new byte[]
-					{
-						(byte)Magic,
-						(byte)(Magic >> 8),
-						(byte)(Magic >> 16),
-						(byte)(Magic >> 24)
-					};
-					_MagicBytes = bytes;
-				}
-				return _MagicBytes;
-			}
-		}
-		public uint Magic
-		{
-			get
-			{
-				return magic;
-			}
-		}
-
-		public Money GetReward(int nHeight)
-		{
-			long nSubsidy = new Money(50 * Money.COIN);
-			int halvings = nHeight / consensus.SubsidyHalvingInterval;
-
-			// Force block reward to zero when right shift is undefined.
-			if(halvings >= 64)
-				return Money.Zero;
-
-			// Subsidy is cut in half every 210,000 blocks which will occur approximately every 4 years.
-			nSubsidy >>= halvings;
-
-			return new Money(nSubsidy);
-		}
-
-		public bool ReadMagic(Stream stream, CancellationToken cancellation, bool throwIfEOF = false)
-		{
-			byte[] bytes = new byte[1];
-			for(int i = 0; i < MagicBytes.Length; i++)
-			{
-				i = Math.Max(0, i);
-				cancellation.ThrowIfCancellationRequested();
-
-				var read = stream.ReadEx(bytes, 0, bytes.Length, cancellation);
-				if(read == 0)
-					if(throwIfEOF)
-						throw new EndOfStreamException("No more bytes to read");
-					else
-						return false;
-				if(read != 1)
-					i--;
-				else if(_MagicBytes[i] != bytes[0])
-					i = _MagicBytes[0] == bytes[0] ? 0 : -1;
-			}
-			return true;
-		}
-	}
+    public class DNSSeedData
+    {
+        private IPAddress[] addresses;
+
+        public string Name { get; }
+
+        public string Host { get; }
+
+        public DNSSeedData(string name, string host)
+        {
+            this.Name = name;
+            this.Host = host;
+        }
+
+        public IPAddress[] GetAddressNodes()
+        {
+            if (this.addresses != null)
+            {
+                return this.addresses;
+            }
+
+            try
+            {
+                this.addresses = Dns.GetHostAddressesAsync(this.Host).Result;
+            }
+            catch(AggregateException ex)
+            {
+                System.Runtime.ExceptionServices.ExceptionDispatchInfo.Capture(ex.InnerException).Throw();
+            }
+
+            return this.addresses;
+        }
+
+        public override string ToString()
+        {
+            return this.Name + " (" + this.Host + ")";
+        }
+    }
+
+    public enum Base58Type
+    {
+        PUBKEY_ADDRESS,
+        SCRIPT_ADDRESS,
+        SECRET_KEY,
+        EXT_PUBLIC_KEY,
+        EXT_SECRET_KEY,
+        ENCRYPTED_SECRET_KEY_EC,
+        ENCRYPTED_SECRET_KEY_NO_EC,
+        PASSPHRASE_CODE,
+        CONFIRMATION_CODE,
+        STEALTH_ADDRESS,
+        ASSET_ID,
+        COLORED_ADDRESS,
+        MAX_BASE58_TYPES,
+    };
+
+    public enum Bech32Type
+    {
+        WITNESS_PUBKEY_ADDRESS,
+        WITNESS_SCRIPT_ADDRESS
```

Comment:
there comes the question - do we want to modify these to comply with our style? XYZ_ABC -> XyzAbc? We probably should ...

Reply:
Not yet, making this change will effect many files.